core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -116,31 +102,33 @@ private void reportCreationError(Exception ex, Uri videoUri) {
     private final MediaPlayer.OnVideoSizeChangedListener internalSizeChangedListener = new MediaPlayer.OnVideoSizeChangedListener() {
         @Override
         public void onVideoSizeChanged(MediaPlayer mp, int width, int height) {
-            if (onSizeChangedListener == null) {
-                return;
+            if (onVideoSizeChangedForwarder == null) {
+                throw new IllegalStateException("Should bind a OnVideoSizeChangedListener. Cannot forward events.");
```

Comment:
For those 3 it is an invalid state to not have a listener ? This class can't be used if we don't care about those events? Just asking, seems ok but wanted to confirm.